client/jetpack-connect/authorize.js

```diff
@@ -283,6 +284,17 @@ export class JetpackAuthorize extends Component {
 		return 'sso' === from && isSsoApproved( clientId );
 	}
 
+	/**
+	 * Check if the user is coming from the Jetpack upgrade flow.
+	 *
+	 * @returns {boolean} True if the user is coming from the Jetpack upgrade flow, false otherwise.
+	 */
+	isJetpackUpgradeFlow() {
+		return this.props.authQuery.redirectAfterAuth.includes(
+			'page=jetpack&action=authorize_redirect'
+		);
+	}
```

Comment:
Looking at the rest of this file, maybe we should follow the same style, something like 

```suggestion
	/**
	 * Check if the user is coming from the Jetpack upgrade flow.
	 *
	 * @param  {object}  props           Props to test
	 * @param  {?string} props.authQuery.redirectAfterAuth Where were we redirected after auth.
	 * @returns {boolean}                True if the user is coming from the Jetpack upgrade flow, false otherwise.
	 */
	isJetpackUpgradeFlow( props = this.props ) {
		const { redirectAfterAuth } = props.authQuery;
		return redirectAfterAuth.includes( 'page=jetpack&action=authorize_redirect' );
	}
```